client/src/components/admin/Dashboard.jsx

```diff
@@ -0,0 +1,124 @@
+import { useEffect } from "react";
+import Sidebar from "./Sidebar";
+import "../../styles/admin/dashboard.css";
+import { Typography } from "@material-ui/core";
+import { Link } from "react-router-dom";
+import { Doughnut, Line } from "react-chartjs-2";
+import { useSelector, useDispatch } from "react-redux";
+import { getAdminProduct } from "../../actions/productAction.js";
+import { getAllOrders } from "../../actions/orderAction.js";
+import { getAllUsers } from "../../actions/userAction.js";
+import MetaData from "../layout/MetaData.jsx";
+import {
+  Chart,
+  CategoryScale,
+  ArcElement,
+  LinearScale,
+  PointElement,
+  LineElement,
+} from "chart.js";
+
+Chart.register(
+  CategoryScale,
+  ArcElement,
+  LinearScale,
+  PointElement,
+  LineElement
+);
+
+const Dashboard = () => {
+  const dispatch = useDispatch();
+
+  const { products } = useSelector((state) => state.products);
+
+  const { orders } = useSelector((state) => state.allOrders);
+
+  const { users } = useSelector((state) => state.allUsers);
+
+  let outOfStock = 0;
+
+  products &&

```

Comment:
use this inside useEffect or any function.

Reply:
resolved